pkg/kube/inject/inject.go

```diff
@@ -464,7 +464,14 @@ func RunTemplate(params InjectionParameters) (mergedPod *corev1.Pod, templatePod
 		// these will be in the `containers` field.
 		// So if we see the proxy container in `containers` in the original pod, and in `initContainers` in the template pod,
 		// move the container.
-		if features.EnableNativeSidecars.Get() &&
+		// The sidecar.istio.io/native-sidecar annotation takes precedence over the global feature flag.
+		native := features.EnableNativeSidecars.Get()
+		if mergedPod.Annotations["sidecar.istio.io/native-sidecar"] == "true" {
```

Comment:
can you add this in API repo first?

Reply:
Ah I didn't realize annotation were in API. Will do.

Reply:
@zirain PTAL at https://github.com/istio/api/pull/3329/files